ext/lib/mgmt/mcumgr/ext/base64/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+add_library(BASE64 INTERFACE)
+
+target_include_directories(BASE64 INTERFACE src)
+
+zephyr_library()
+target_include_directories(BASE64 INTERFACE
+    include
+    port/zephyr/include
+)
+
+zephyr_library_sources(
+    src/base64.c
```

Comment:
base64 is already in mbedtls. for me it is ok to have both in zephyr - but i was criticized while I want to adapt exactly these mynewt base64 sources for zephyr. For instance mcuboot serial recover is using mbdtls-base64 lib.

Reply:
I was a bit wary of depending on mbedtls just for base64 support.  However, on reflection I agree that it is better than having two base64 libraries.

Reply:
One issue with using mbedtls is that it is difficult to enable base64 support.  Most of mbedtls is configured via a header file rather than Kconfig, so it is not possible to automatically enable a single setting.  

By default, mbedtls is configured to use as `ext/lib/crypto/mbedtls/configs/config-threadnet.h` as its configuration file, which does not enable base64 support.  This header file is part of the mbedtls distribution, so it is not something we should change just to make it easier to include mcumgr in a build.

The point is: an mcumgr user will need to configure mbedtls to use a different header file, or else the build will fail with linker errors.  This seems like a lousy enough user experience that I am holding off on pushing the base64 changes for now.  If the consensus is to take out the additional base64 library and use mbedtls, then I will push the changes.

Reply:
Right - so it is better to have mcumgr-base64 here. Another advantage of this is that we have more control on mcumghr than on mbedtls (which is ARM project).

Reply:
I went ahead and removed the base64 library (along with the `ext/` directory) from mcumgr.  Now, Zephyr uses mbedTLS's base64 support when using mcumgr.

I enabled base64 support by default with a modification to `ext/lib/crypto/mbedtls/configs/config-threadnet.h` (https://github.com/zephyrproject-rtos/zephyr/pull/6068/commits/bf3bcf5fd942c84922a96ab9347a3db455154d4b).